extension/healthcheckv2extension/extension.go

```diff
@@ -7,36 +7,198 @@ import (
 	"context"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
 	"go.opentelemetry.io/collector/extension"
+	"go.uber.org/multierr"
```

Comment:
we are not using this anymore, preferring the similar feature from the stdlib: https://pkg.go.dev/errors#Join